### ISSUE TITLE
Can't find my tessel over wlan

### ISSUE THREAD

**Issue by LinusU** [tessel/t2-cli#190]
My computers builtin mdns discovery tool finds it.

```text
➜  ~  dns-sd -B _tessel._tcp .
Browsing for _tessel._tcp
DATE: ---Tue 07 Jul 2015---
18:47:19.924  ...STARTING...
Timestamp     A/R    Flags  if Domain               Service Type         Instance Name
18:47:19.926  Add        2   4 local.               _tessel._tcp.        frog
```

But it dosen't show up in the cli, even with a higher timeout.

```text
➜  t2-cli git:(master) t2 list --timeout 10
INFO Searching for nearby Tessels...
	frog	USB	
```

**Comment by johnnyman727** [comment]
@LinusU it would be useful to know if [this line ever gets hit](https://github.com/tessel/t2-cli/blob/master/lib/lan_connection.js#L132) and if any devices get emitted [here](https://github.com/tessel/t2-cli/blob/master/lib/lan_connection.js#L139).

**Comment by LinusU** [comment]
I've found the problem, will post writeup soon

**Comment by johnnyman727** [comment]
:racehorse:

**Comment by LinusU** [comment]
Hehe, basically this is the problem: kmpm/node-mdns-js#40

**Comment by johnnyman727** [comment]
Fascinating, what is the error that gets swallowed?

**Comment by LinusU** [comment]
> Error: ENOENT, no such file or directory '/Users/linus/.tessel/id_rsa'

Eh, I have no idea how but now I suddenly have that file, it's like it has run `t2 provision` in the background... Is it suppose to do that?

**Comment by johnnyman727** [comment]
That's really strange - I *don't think* `provision` ever gets called in the background. @rwaldron may have had this issue before as well.

**Comment by LinusU** [comment]
There is also an interesting race condition that happens rarely. It will find my tessel (`connection` gets emitted) but it aborts the scan before `tessel.connection.open` is done. This makes it so that the `tessel` event never fires and it errs out with `No tessels found`.

**Comment by LinusU** [comment]
I think that we can call this resolved by #192 since that would have made the error vey apparent.

**Comment by johnnyman727** [comment]
> There is also an interesting race condition that happens rarely. It will find my tessel (connection gets emitted) but it aborts the scan before tessel.connection.open is done.

Yes, I noticed this as well. As a shitty workaround, #182 increases the default timeout from 3s to 5s but it does resolve the problem. The more comprehensive solution would obviously be to wait for any pending connections to complete before ending the process.

**Comment by Frijol** [comment]
I'm reasonably certain `t2 provision` doesn't get called in the background anywhere, but it was discussed in some length https://github.com/tessel/t2-cli/issues/64

**Comment by johnnyman727** [comment]
Closed with #182.
